public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -294,7 +295,8 @@ export class LokiDatasource
   query(request: DataQueryRequest<LokiQuery>): Observable<DataQueryResponse> {
     const queries = request.targets
       .map(getNormalizedLokiQuery) // used to "fix" the deprecated `.queryType` prop
-      .map((q) => ({ ...q, maxLines: q.maxLines ?? this.maxLines }));
+      .map((q) => ({ ...q, maxLines: q.maxLines ?? this.maxLines }))
+      .map(removeInvalidOptions);
```

Comment:
Why isn't this in `getNormalizedLokiQuery` and move `.map((q) => ({ ...q, maxLines: q.maxLines ?? this.maxLines })` as `getNormalizedLokiQuery`  seems like a perfect place for this.

Reply:
I went ahead and made this change